public/main.js

```diff
@@ -117,53 +151,18 @@ function findTodo(id) {
     }
 }
 
-function updateTodo(element, todoId, callback) {
-    var updateRequest = new XMLHttpRequest();
-    var  todo = findTodo(todoId);
-    updateRequest.open("PUT", "/api/todo/" + todo.id);
-    updateRequest.setRequestHeader("Content-type", "application/json");
-    var d = element.checked;
-    updateRequest.send(JSON.stringify({
-        title: todo.title,
-        isComplete: element.checked
-    }));
-    updateRequest.onload = function() {
-        if (this.status === 200) {
-            if (element.checked) {
-                element.parentNode.className += "completed";
-            } else {
-                element.parentNode.className = element.parentNode.className.replace("completed", "");
-            }
-            callback();
-        } else {
-            renderMessageDialog("error", "Failed to update item " +
-                todoId.toString() + ". Server returned " + this.status + " - " + this.responseText);
-        }
-    };
-}
-
-function deleteTodo(todoId, callback) {
-    var deleteRequest = new XMLHttpRequest();
-    deleteRequest.open("DELETE", "/api/todo/" + todoId);
-    deleteRequest.onload = function() {
-        if (this.status === 200) {
-        } else {
-            renderMessageDialog("error", "Failed to delete item " +
-                todoId.toString() + ". Server returned " + this.status + " - " + this.responseText);
-        }
-        callback(this.status);
-    };
-    deleteRequest.send();
+function checkStatus(response) {
+    if (response.status >= 200 && response.status < 300) {
```

Comment:
Strictly speaking, 300 codes aren't errors.
Also, and I might be wrong on this one, if you do get a 400 or 500 error code, this will be caught by fetch and your 'catch' clause will be called, so I am not sure this checkstatus method is really needed.